contracts/verifiers/Verifier.sol

```diff
@@ -323,20 +317,33 @@ abstract contract Verifier is IVerifier, ContextUpgradeable {
         $._state.processCrossChainProofs(crossChainProofs);
 
         uint256 userIDFromAuthResponse;
-        AuthTypeData storage authTypeData = $._authMethods[authResponse.authType];
+        AuthMethodData storage authMethodData = $._authMethods[authResponse.authMethod];
 
         // 2. Authenticate user and get userID
-        bytes32 expectedNonce = keccak256(abi.encode(sender, responses)) &
-            0x0FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF;
-
-        userIDFromAuthResponse = authTypeData.validator.verify(
-            authResponse.proof,
-            authTypeData.params,
+        IAuthValidator.AuthResponseField[] memory authResponseFields;
+        (userIDFromAuthResponse, authResponseFields) = authMethodData.validator.verify(
             sender,
-            $._state,
-            expectedNonce
+            authResponse.proof,
+            authMethodData.params
         );
 
+        if (
+            keccak256(abi.encodePacked(authResponse.authMethod)) ==
+            keccak256(abi.encodePacked("authV2"))
```

Comment:
Maybe we could have it precalculated as a constant.

Reply:
Done